File: src/BjjEire.Domain/Entities/Stores/Store.cs
// Copyright (c) BjjEire. All rights reserved.
// Licensed under the MIT License.


namespace BjjEire.Domain.Entities.Stores;

public class Store : BaseEntity
{
    [BsonElement("name")]
    public string Name { get; set; } = string.Empty;

    [BsonElement("description")]
    public string? Description { get; set; }

    [BsonElement("websiteUrl")]
    public string WebsiteUrl { get; set; } = string.Empty;

    [BsonElement("logoUrl")]
    public string? LogoUrl { get; set; }

    [BsonElement("isActive")]
    public bool IsActive { get; set; } = true;

}

File: src/bjjeire-api/src/main/java/com/bjjeire/api/store/Store.java
package com.bjjeire.api.store;

import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(Store.ENTITY_NAME)
public class Store {
    public static final String ENTITY_NAME = "Store";

    @Id
    private String id;

    private String name;
    private String description;
    private String websiteUrl;
    private String logoUrl;

    @Builder.Default
    private boolean isActive = true;

    @Field("createdAt")
    private Instant createdOnUtc;

    @Field("createdBy")
    private String createdBy;

    @Field("updatedAt")
    private Instant updatedOnUtc;

    @Field("updatedBy")
    private String updatedBy;
}
